charms/kubernetes_snaps.py

```diff
@@ -433,7 +467,106 @@ def configure_scheduler(extra_args_config, kubeconfig):
     with open(kube_scheduler_config_path, "w") as f:
         yaml.safe_dump(scheduler_config, f)
 
-    configure_kubernetes_service("kube-scheduler", scheduler_opts, extra_args_config)
+    config_files = {
+        scheduler_opts["config"],
+        kubeconfig,
+        kube_scheduler_config_path,
+    }
+
+    configure_kubernetes_service(
+        "kube-scheduler", scheduler_opts, extra_args_config, config_files
+    )
+
+
+def _sha256_file(config_file: str) -> hashlib.sha256:
+    h = hashlib.sha256()
+    h.update(config_file.encode())
+    config_file = Path(config_file)
+    if Path.is_file(config_file):
+        with config_file.open("rb") as file:
+            while True:
+                # Reading is buffered, so we can read smaller chunks.
+                chunk = file.read(h.block_size)
+                if not chunk:
+                    break
+                h.update(chunk)
+    else:
+        h.update(b"--empty--")
+    return h
+
+
+def _dict_compare(d1, d2):
+    d1_keys = set(d1.keys())
+    d2_keys = set(d2.keys())
+    shared_keys = d1_keys.intersection(d2_keys)
+    added = d1_keys - d2_keys
+    removed = d2_keys - d1_keys
+    modified = {o: (d1[o], d2[o]) for o in shared_keys if d1[o] != d2[o]}
+    same = set(o for o in shared_keys if d1[o] == d2[o])
+    return added, removed, modified, same
```

Comment:
Just a nit to make the function more generic. I think `o` is short for option, right?
```suggestion
def _dict_compare(d1, d2):
    d1_keys = set(d1.keys())
    d2_keys = set(d2.keys())
    shared_keys = d1_keys & d2_keys
    added = d1_keys - d2_keys
    removed = d2_keys - d1_keys
    modified = {k: (d1[k], d2[k]) for k in shared_keys if d1[k] != d2[k]}
    same = {k for k in shared_keys if d1[k] == d2[k]}
    return added, removed, modified, same
```

Reply:
good suggestion!!